elgamal/ciphertext_test.go

```diff
@@ -139,3 +153,134 @@ func encrypt(message *big.Int, publicKey *babyjub.PublicKey, k *big.Int) (*babyj
 	c2p := babyjub.NewPointProjective().Add(m.Projective(), s.Projective())
 	return c1, c2p.Affine()
 }
+
+type testElGamalEncryptCircuit struct {
+	PrivKey frontend.Variable
+	PubKey  twistededwards.Point `gnark:",public"`
+	Result  Ciphertext           `gnark:",public"`
+	K       frontend.Variable
+	Msg     frontend.Variable
+}
+
+func (c *testElGamalEncryptCircuit) Define(api frontend.API) error {
+	api.Println(c.PrivKey)
+	res, err := new(Ciphertext).Encrypt(api, c.PubKey, c.K, c.Msg)
+	if err != nil {
+		return err
+	}
+	res.AssertIsEqual(api, &c.Result)
+	return res.AssertDecrypt(api, c.PrivKey, c.Msg)
+}
+
+func TestEncryptAssertDecrypt(t *testing.T) {
+	// generate a public mocked key and a random k to encrypt first message
+	privKey, pubKey, err := generateKeyPair(nil)
+	if err != nil {
+		t.Fatalf("Error generating key pair: %v\n", err)
+		return
+	}
+	k, err := randomK()
+	if err != nil {
+		t.Fatalf("Error generating random k: %v\n", err)
+		return
+	}
+	// encrypt a simple message
+	msg := big.NewInt(3)
+	a1, a2 := encrypt(msg, pubKey, k)
+	// reduce the points to reduced twisted edwards form
+	xA1RTE, yA1RTE := format.FromTEtoRTE(a1.X, a1.Y)
```

Comment:
Why is this? Are we using iden3/bjj somewhere? Would it make sense to user gnark everywhere?

Reply:
Every test of this package is using the iden3 bjj, I prefer to not change it to use the same helpers.